### PR TITLE
Fix wasm mmap

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mono/utils/mono-mmap-wasm.c [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+	if (!malloced_shared_area)
+		malloced_shared_area = mono_malloc_shared_area (getpid ());
+	/* get the pid here */
+	return malloced_shared_area;
+}
+
+void
+mono_shared_area_remove (void)
+{
+	if (malloced_shared_area)
```
Comment (jaykrell) [inline mono/utils/mono-mmap-wasm.c]: No need for `if`.

File: mono/utils/mono-mmap-wasm.c [hunk 2]
```diff
@@ -0,0 +1,223 @@
+
+#include <config.h>
+
+#ifdef HOST_WASM
+
+#include <sys/types.h>
+#if HAVE_SYS_STAT_H
```
Comment (jaykrell) [inline mono/utils/mono-mmap-wasm.c]: Can likely dial down the `#if` for wasm.

**Comment by vargaz** [comment]
Waiting to see whenever this really fixes things, don't merge yet.

**Comment by vargaz** [comment]
This seems to cause random crashes which are very hard to debug.
Closed for now in favour of:
https://github.com/mono/mono/pull/11840

**Pull request closed without merging**